### PR TITLE
Refs #36665 - Remove katello-agent end points

### PR DESCRIPTION
**Pull request by parthaa**
This PR removes all the katello-agent endpoint  related hammer actions. Instead of saying `no such command` it tries to redirect the use to the right area.

```bash
$ hammer host package
Usage:
    hammer host package [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    Subcommand
 [ARG] ...                     Subcommand arguments

Subcommands:
 install                       Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_install`.
 list                          List packages installed on the host
 remove                        Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_remove`.
 upgrade                       Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_update`.
 upgrade-all                   Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_update`.

Options:
 -h, --help                    Print help

$ hammer host package install --id=1000 
Not supported. Use the remote execution equivalent `hammer job-invocation create --feature katello_package_install`.:
  Error: Unrecognised option '--id'.
  
  See: 'hammer host package install --help'.

$ hammer host package install 
Not supported. Use the remote execution equivalent `hammer job-invocation create --feature katello_package_install`.

$ hammer host package install -h  
Not supported. Use the remote execution equivalent `hammer job-invocation create --feature katello_package_install`.

$ hammer host-collection
Usage:
    hammer host-collection [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    Subcommand
 [ARG] ...                     Subcommand arguments

Subcommands:
.....
 erratum                       Manage errata on your host collections. Note these command are no longer
                               available. Use the remote execution equivalent
 package                       Manage packages on your host collections. Note these command are no longer
                               available. Use the remote execution equivalent
 package-group                 Manage package-groups on your host collections. Note these command are no longer
                               available. Use the remote execution equivalent
....
Options:
 -h, --help                    Print help

$  hammer host-collection package --help
Usage:
    hammer host-collection package [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    Subcommand
 [ARG] ...                     Subcommand arguments

Subcommands:
 install                       Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_install`. Specify the host collection with the
                               --search-query parameter, e.g. `--search-query "host_collection = MyCollection"`
                               or `--search-query "host_collection_id=6"`
 remove                        Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_remove`. Specify the host collection with the
                               --search-query parameter, e.g. `--search-query "host_collection = MyCollection"`
                               or `--search-query "host_collection_id=6"`
 update                        Not supported. Use the remote execution equivalent `hammer job-invocation create
                               --feature katello_package_update`. Specify the host collection with the
                               --search-query parameter, e.g. `--search-query "host_collection = MyCollection"`
                               or `--search-query "host_collection_id=6"`

Options:
 -h, --help                    Print help

$ hammer host-collection package install --help

Not supported. Use the remote execution equivalent `hammer job-invocation create --feature katello_package_install`. Specify the host collection with the --search-query parameter, e.g. `--search-query "host_collection = MyCollection"` or `--search-query "host_collection_id=6"`
```

### CODE REVIEW

File: lib/hammer_cli_katello/host_package_group.rb [hunk 1]
```diff
@@ -1,48 +1,22 @@
...
-        option(:option_groups).required
-      end
-
-      build_options :without => [:packages]
-
       def execute
-        warn "This command uses katello agent and will be removed in favor of remote execution " \
-          "in Katello 4.10."
-        warn "The remote execution equivalent is `hammer job-invocation create --feature " \
+        warn "Use the remote execution equivalent `hammer job-invocation create --feature " \
```
Comment (ekohl) [inline lib/hammer_cli_katello/host_package_group.rb:9]: Should this also be `fail`?
```suggestion
        fail "Use the remote execution equivalent `hammer job-invocation create --feature " \
```

File: lib/hammer_cli_katello/host_package_group.rb [hunk 2]
```diff
@@ -1,48 +1,22 @@
...
-        option(:option_groups).required
-      end
-
-      build_options :without => [:packages]
 
       def execute
-        warn "This command uses katello agent and will be removed in favor of remote execution " \
-          "in Katello 4.10."
-        warn "The remote execution equivalent is `hammer job-invocation create --feature " \
+        warn "Use the remote execution equivalent is `hammer job-invocation create --feature " \
```
Comment (ekohl) [inline lib/hammer_cli_katello/host_package_group.rb:18]: And this also `fail`?
```suggestion
        fail "Use the remote execution equivalent is `hammer job-invocation create --feature " \
```

**Review by jeremylenz — APPROVED** [review]
I'm good with this approach. :+1: 

APJ

File: lib/hammer_cli_katello/host_collection_erratum.rb [hunk 3]
```diff
@@ -1,33 +1,17 @@
 module HammerCLIKatello
   class HostCollectionErratumCommand < HammerCLIKatello::Command
     command_name "erratum"
-    desc _("Manipulate errata for a host collection")
+    desc "Manage errata on your host collections. "\
+         "Note these command are no longer available. "\
```
Comment (jeremylenz) [inline lib/hammer_cli_katello/host_collection_erratum.rb:5]:
```suggestion
         "These commands are no longer available. "\
```

Comment (jeremylenz) [reply lib/hammer_cli_katello/host_collection_erratum.rb:5]: same for all other occurrences

**Review by jeremylenz — APPROVED** [review]

**Pull request merged**